src/rust/types/src/lib.rs

```diff
@@ -0,0 +1,133 @@
+use bigdecimal::{BigDecimal, ToPrimitive};
+use chrono::{DateTime, Utc};
+use error::TypeError;
+use serde::{Deserialize, Serialize};
+
+pub mod error;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Coin {
+    pub account_address: String,
+    pub module_name: String,
+    pub struct_name: String,
+    pub symbol: String,
+    pub name: String,
+    pub decimals: i16,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Market {
+    pub market_id: u64,
+    pub base: Coin,
+    pub base_name_generic: Option<String>,
+    pub quote: Coin,
+    pub lot_size: u64,
+    pub tick_size: u64,
+    pub min_size: u64,
+    pub underwriter_id: u64,
+    pub created_at: DateTime<Utc>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct QueryMarket {
+    pub market_id: BigDecimal,
+    pub base_symbol: Option<String>,
```

Comment:
Should be some only if base asset is a coin

Reply:
Yup `base` on `Market` should be an `Option<Coin>`